### PR TITLE
feat: add EngineApiKind enum

### PR DESCRIPTION
**Pull request by mvares**
Partially #10124

### CODE REVIEW

File: crates/chainspec/src/spec.rs [hunk 1]
```diff
@@ -258,6 +258,13 @@ impl From<ForkBaseFeeParams> for BaseFeeParamsKind {
     }
 }
 
+/// The type for specify which is kind of engine api
+#[derive(Debug)]
```
Comment (mattsse) [inline crates/chainspec/src/spec.rs:262]: also add copy, clone, partialeq, eq

File: crates/chainspec/src/spec.rs [hunk 2]
```diff
@@ -258,6 +258,13 @@ impl From<ForkBaseFeeParams> for BaseFeeParamsKind {
...
+pub enum EngineApiKind {
+    Ethereum,
+    OpStack,
```
Comment (mattsse) [inline crates/chainspec/src/spec.rs:265]: one line docs

File: crates/chainspec/src/spec.rs [hunk 3]
```diff
@@ -258,6 +258,13 @@ impl From<ForkBaseFeeParams> for BaseFeeParamsKind {
     }
 }
 
+/// The type for specify which is kind of engine api
+#[derive(Debug)]
+pub enum EngineApiKind {
+    Ethereum,
```
Comment (mattsse) [inline crates/chainspec/src/spec.rs:264]: this should be default

File: crates/chainspec/src/spec.rs [hunk 4]
```diff
@@ -258,6 +258,13 @@ impl From<ForkBaseFeeParams> for BaseFeeParamsKind {
     }
 }
 
+/// The type for specify which is kind of engine api
+#[derive(Debug)]
+pub enum EngineApiKind {
```
Comment (mattsse) [inline crates/chainspec/src/spec.rs:263]: this should be moved to:

https://github.com/paradigmxyz/reth/blob/06e7afbd2d898b6f95ff7b757f53327d189e2048/crates/engine/tree/src/engine.rs

File: crates/chainspec/src/spec.rs [hunk 5]
```diff
@@ -366,6 +373,15 @@ impl ChainSpec {
...
     }
 
+    #[inline]
+    /// Returns [`EngineApiKind`]
+    pub const fn get_engine_api_kind(&self) -> EngineApiKind {
+        match self.chain.is_optimism() {
+            true => EngineApiKind::OpStack,
+            false => EngineApiKind::Ethereum,
+        }
+    }
```
Comment (mattsse) [inline crates/chainspec/src/spec.rs:383]: this we don't need

**Review by mattsse — CHANGES_REQUESTED** [review]
ty, some suggestions

File: crates/chainspec/src/spec.rs (hunk 4 above)
Comment (mvares) [reply crates/chainspec/src/spec.rs:263]: @mattsse, what does you means by `defaut`?

File: crates/chainspec/src/spec.rs (hunk 3 above)
Comment (mvares) [reply crates/chainspec/src/spec.rs]: @mattsse, what does you means by defaut?

**Review by mattsse — APPROVED** [review]

**Pull request merged**